### ISSUE TITLE
1.15.2 forge 31.1.74 ticking entity/ enum crash

### ISSUE THREAD

**Issue by ArMoReDfUrY** [DeflatedPickle/Heart-Drops#5]
hi, this may be similar to #4 but i think it may have something to do with recent forge change to enum values.

https://pastebin.com/g87XygAY

EDIT: i am using default config values

**Comment by DeflatedPickle** [comment]
The link has been wiped now, so I can't check the stack trace, but I don't think it's related to #4. Pastebin removes pastes real quick, I think Hatebin keeps them for longer?

I've had this before but someone wanted a 1.15 version and the error didn't pop up during the test run, so I shipped it. I did ask around on MMD, but nobody was able to help.

**Comment by ArMoReDfUrY** [comment]
that was me who asked :) Fury1755. my git name is different than my curse name.
im about to test the new version 3.0.0.4 now. will let you know how it goes.

**Comment by DeflatedPickle** [comment]
Ah. I should have guessed by the picture. I don't think the new version will fix this issue, though.

**Comment by ArMoReDfUrY** [comment]
Build: 1.15.2-31.1.64 - Sun May 03 19:56:55 GMT 2020
	ichttt:
		Fix config comments referencing disallowed enum values. (#6542)

that is the build with the info update although im not sure when they made the change.

**Comment by DeflatedPickle** [comment]
It's a problem with config *comments*?

**Comment by ArMoReDfUrY** [comment]
no. thats when they updated the information for enum formatting

**Comment by DeflatedPickle** [comment]
The commit title suggests it's about config comments, though.

**Comment by ArMoReDfUrY** [comment]
it is. hmm my pastebin has expired. give me a few minutes to replicate the issue and i'll see if i can gleam a bit more info.

**Comment by ArMoReDfUrY** [comment]
well i cant seem to get it to crash on 3.0.0.4.
i have killed about 30+ mobs after letting them bring me down to 1 heart and i have not had a single heart drop...
this is with the default config.
i am puzzled

**Comment by ArMoReDfUrY** [comment]
and it just crashed on it's own. gathering the report now

The game crashed whilst ticking entity
Error: java.lang.ClassCastException: Cannot convert an Integer to an Enum: disallowed by EnumGetMethod.NAME_IGNORECASE

**Comment by ArMoReDfUrY** [comment]
https://pastebin.com/kPsw4pdx

**Comment by ArMoReDfUrY** [comment]
i will try replicate it on a clean setup only using heartdrops.
let you know soon.

**Comment by ArMoReDfUrY** [comment]
ok here is the report for just heart drops by itself
https://pastebin.com/xwZM2chr

**Comment by DeflatedPickle** [comment]
Right. Well. In respect to the items not dropping, it's possible the default config is broken/I've done something stupid. The only changes were checking if the source is an entity before using it as one, how'd that break it?! We can't debug that with crash logs, that'll have to be run through a debugger with breakpoints.

As for that crash log, that's the error I was getting. I don't get what it's talking about. I don't get what it's thinking is an integer and what enum it actually is.

My recommendation at this point; don't use this mod. I don't know how to fix this, I don't know what to debug.

**Comment by ArMoReDfUrY** [comment]
ok i think i've figured it out.

i have changed the config setting
	#Changes when the hearts will drop
	#Allowed Values: HURT, ALWAYS, NEVER
	dropWhen = "ALWAYS"

and now it works without issue. hearts drop. no crashes

according to the Logs, Line 109 of src/main/kotlin/com/deflatedpickle/heartdrops/event/ForgeEventHandler.kt
calls on the config file here,
        107            when (Config.dropWhen.get()!!) {
        108                When.HURT -> { 
this>109                    amount = when (Config.dropAmount.get()!!) {

this integer for percentage of mob health seems to be the cause.
why... i have no idea, maybe you can bake that in so it doesn't call on the config for that integer.

for now i'll just leave it set to "ALWAYS" . this should drop 1+1/2 hearts per mob kill without issue.

maybe change the default config setting to "ALWAYS" for 1.15.x as a temp solution?

**Comment by DeflatedPickle** [comment]
I actually just ran it to test and I was just about to suggest that. The only problem with it is it then no longer checks a box around the entity to find hearts on the floor so it can work out how many to drop. That's only available with `dropWhen = HURT`.

Alright, let's all collectively laugh at the dumbie dumb dumb mod dev. I set the config keys as the same for the `dropAmount` and the `dropAmountValue`. No wonder I was getting this error.

I'll see if I can fix the pre-set setting for `dropWhen`, though. I like it dropping just enough to heal you.
